src/main/java/org/jenkinsci/plugins/github/extension/GHEventsSubscriber.java

```diff
@@ -55,12 +55,13 @@
      * This method called when root action receives webhook from GH and this extension is interested in such
      * events (provided by {@link #events()} method). By default do nothing and can be overrided to implement any
      * parse logic
-     * Don't call it directly, use {@link #processEvent(GHEvent, String)} static function
+     * Don't call it directly, use {@link #processEvent(GHEvent, String, String)} static function
```

Comment:
I'm mostly thinking about risks outside the CI environment, for example if an attacker obtains a secret, it would be possible to track down which project the secret belongs to. Having project-specific secrets would also prevent them from starting to execute jobs belonging to other project as the key is tied to a specific project. From my point of view, it is not the issue that an admin has access to the keys for all the projects, but that the keys are loosely tied to the projects when they are system-wide and if there is a large amount of hooks that have different repos with different repo owners.